EasyPost/PaymentMethod.cs

```diff
@@ -1,38 +1,119 @@
-        public CreditCard primary_payment_method { get; set; }
+        public PaymentMethodObject primary_payment_method { get; set; }
         [JsonProperty("secondary_payment_method")]
-        public CreditCard secondary_payment_method { get; set; }
+        public PaymentMethodObject secondary_payment_method { get; set; }
+    }
+
+    /// <summary>
+    ///     Represents a credit card or a bank account.
+    ///     Have to collect all possible properties since we can't anticipate which type of payment method the API will return.
+    /// </summary>
+    public class PaymentMethodObject : Resource
+    {
+        public enum PaymentMethodType
+        {
+            CreditCard,
+            BankAccount
+        }
+
+        // bank_account
+        [JsonProperty("bank_name")]
+        public string bank_name { get; set; }
+        // credit_card
+        [JsonProperty("brand")]
+        public string brand { get; set; }
+        // bank_account
+        [JsonProperty("country")]
+        public string country { get; set; }
+        // both
+        [JsonProperty("disabled_at")]
+        public string disabled_at { get; set; }
+        // credit_card
+        [JsonProperty("exp_month")]
+        public string exp_month { get; set; }
+        // credit_card
+        [JsonProperty("exp_year")]
+        public string exp_year { get; set; }
+        // both
+        [JsonProperty("id")]
+        public string id { get; set; }
+        // both
+        [JsonProperty("last4")]
+        public string last4 { get; set; }
+        // credit_card
+        [JsonProperty("name")]
+        public string name { get; set; }
+        // both
+        [JsonProperty("object")]
+        public string Object { get; set; }
 
         /// <summary>
-        ///     List all payment methods for this account.
+        ///     Get what type of payment method this is (credit card, bank account, etc.)
         /// </summary>
-        /// <returns>An EasyPost.PaymentMethod summary object.</returns>
-        /// <exception cref="Exception"></exception>
-        public static async Task<PaymentMethod> All()
+        public PaymentMethodType? Type
         {
-            Request request = new Request("payment_methods", Method.Get);
+            get
+            {
+                if (id == null)
+                {
+                    return null;
+                }
 
-            PaymentMethod paymentMethod = await request.Execute<PaymentMethod>();
+                if (id.StartsWith("card_"))
+                {
+                    return PaymentMethodType.CreditCard;
+                }
 
-            if (paymentMethod.id == null)
-            {
-                throw new Exception("Billing has not been setup for this user. Please add a payment method.");
+                if (id.StartsWith("bank_"))
+                {
+                    return PaymentMethodType.BankAccount;
+                }
+
+                return null;
```

Comment:
Issue: Only one return statement please instead of 4 in one function. To Jack's point, I feel there is potential for some refactoring here to clean it up a bit.

Reply:
I'm confused why this is giving you pause. This is effectively a switch statement, returning a value based on what the ID prefix is: `CreditCard` if it's a credit card, `BankAccount` if it's a bank account or `null` if the ID either is not set or is not one of the two expected ID types.